docker/generic/start_proxy.py

```diff
@@ -1011,12 +1010,14 @@ def make_argparser():
 # Check whether there are conflict flags. If so, return the error string.
 # Otherwise returns None. This function also changes some default flag value.
 def enforce_conflict_args(args):
-    if args.rollout_strategy:
-        if args.rollout_strategy != DEFAULT_ROLLOUT_STRATEGY:
-          if args.version:
-            return "Flag --version cannot be used together with -R or --rollout_strategy."
-          if args.service_json_path:
+    if args.rollout_strategy == "managed":
+        if args.version:
+            return "Flag --version cannot be used if --rollout_strategy=managed."
+        if args.service_json_path:
             return "Flag -R or --rollout_strategy must be fixed with --service_json_path."
+    else:
+        if not args.version and not args.service_json_path:
```

Comment:
no need check `not args.service_json_path`?

Reply:
Yes, we need to check.  If service_json_path is specified,  version is NOT required.